File: app/src/main/java/app/endershrooms/thatcatapp/model/helper/IntToBooleanAdapter.java
package app.endershrooms.thatcatapp.model.helper;

import com.squareup.moshi.FromJson;
import com.squareup.moshi.ToJson;

public class IntToBooleanAdapter {
  @FromJson
  @IntToBoolean
  boolean fromJson(int value) {
    return value != 0;
  }

  @ToJson
  int toJson(@IntToBoolean boolean value) {
    return value ? 1 : 0;
  }
}

File: app/src/main/java/app/endershrooms/thatcatapp/model/helper/IntToBooleanAdapter.kt
package app.endershrooms.thatcatapp.model.helper

import com.squareup.moshi.FromJson
import com.squareup.moshi.ToJson

class IntToBooleanAdapter {
    @FromJson
    @IntToBoolean
    fun fromJson(value: Int): Boolean = value != 0

    @ToJson
    fun toJson(@IntToBoolean value: Boolean): Int = if (value) 1 else 0
}